bin-src/io/HTTPClient.ts

```diff
@@ -54,11 +63,28 @@ export default class HTTPClient {
   }
 
   async fetch(url: string, options: RequestInit = {}, opts: HTTPClientFetchOptions = {}) {
-    const agent = options.agent || getProxyAgent({ env: this.env, log: this.log }, url, opts.proxy);
+    let agent = options.agent || getProxyAgent({ env: this.env, log: this.log }, url, opts.proxy);
+
+    if (this.env.CHROMATIC_DNS_SERVERS.length) {
+      this.log.debug(`Using custom DNS servers: ${this.env.CHROMATIC_DNS_SERVERS.join(', ')}`);
+      dns.setServers(this.env.CHROMATIC_DNS_SERVERS);
+      agent = getCustomDNSAgent();
```

Comment:
It's either/or. If they specify `CHROMATIC_DNS_SERVERS` then we won't use a potential proxy server. That should be made clear in the docs.

Reply:
Should we base it on the `cURL` behavior where proxy setting takes priority instead of custom DNS?

Reply:
In hindsight, I was wondering the same thing (was chatting with Cody about this). In most setups where a proxy is used, the client DNS is secondary since the proxy server will be responsible for doing the DNS queries.

Reply:
Done: https://github.com/chromaui/chromatic-docs/pull/191